baseapp/baseapp.go

```diff
@@ -70,15 +71,16 @@ func (app *BaseApp) Name() string {
 }
 
 // Mount a store to the provided key in the BaseApp multistore
+// Broken until #532 is implemented.
 func (app *BaseApp) MountStoresIAVL(keys ...*sdk.KVStoreKey) {
 	for _, key := range keys {
-		app.MountStore(key, sdk.StoreTypeIAVL)
+		app.MountStore(key, sdk.StoreTypeIAVL, app.db)
 	}
 }
 
 // Mount a store to the provided key in the BaseApp multistore
-func (app *BaseApp) MountStore(key sdk.StoreKey, typ sdk.StoreType) {
-	app.cms.MountStoreWithDB(key, typ, app.db)
+func (app *BaseApp) MountStore(key sdk.StoreKey, typ sdk.StoreType, db dbm.DB) {
```

Comment:
let's leave the old MountStore, mark it as broken like above, and add a new MountStoreWithDB method to BaseApp